### ISSUE TITLE
Laravel Signed URL not respecting HTTPS

### ISSUE THREAD

**Issue by andrewgosselin** [laravel/framework#31430]
- Laravel Version: 6.2
- PHP Version: 7.2
- Database Driver: MySQL

### Description:
I have had a constant issue when using Signed URL's in my applications while trying to force HTTPS, further investigation it seems the UrlGenerator class is getting 2 different signatures when I am using `\URL::forceSheme("https");` in my AppServiceProviders boot method.

It seems to be generating the url with an insecure url and comparing it with a secure url which results in a 403 invalid signature.

### Steps To Reproduce:
1) Set your .env APP_URL to your secure URL. ( I have tried both http and https in the env without change. )
2) Set your .env APP_ENV to `test`.
3) Put the following code into the boot method of your AppServiceProvider.
```php
$environments = ['test'];
if (in_array(env('APP_ENV'), $environments)) {
        \URL::forceScheme('https');
}
```
4) Generate and navigate to a signed URL.

5) You will get a 403 Invalid signature.

**Comment by driesvints** [comment]
Almost all of the previous reports for this were caused by improper configured trusted proxies.

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to report back and I'll gladly help you out and re-open this issue.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

Thanks!

**Comment by andrewgosselin** [comment]
I have already tried most of what I could find on most of those sites beforehand, I am not understanding how this could be related to trust proxies? It works when I am not doing a forceScheme, its not related to the headers being tampered with, its just not giving a secure URL on Laravels end and besides I am trusting "*" so this shouldn't be the issue.

**Comment by andrewgosselin** [comment]
Also this seems to be an issue that has been closed without solution a lot.

https://github.com/laravel/framework/issues/26834

Mainly the one above, the guy created a pull request but it was declined due to not being tested and not having enough explanation. (https://github.com/laravel/framework/pull/27699)

This issue seems like the same issue that I am having, I dont believe this is in any way misconfiguration. I have also made a post on a work account about this issue with no resolution.

This could be caused by our ssl being done on our proxy? but if thats the case Laravel is not handling this correctly since it should be using the url specified in the .env.

Theoretically if my .env domain is https:// then the url generated by laravel in this situation should also be https:// but it is not.
